src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -455,43 +457,125 @@ def create_resource(properties,
     return res.create_resource(properties, location, is_full_object)
 
 
-def show_resource(resource_group_name=None,
+def show_resource(resource_ids=None, resource_group_name=None,
                   resource_provider_namespace=None, parent_resource_path=None, resource_type=None,
                   resource_name=None, api_version=None):
+    if resource_ids:
+        results = []
+        for rid in resource_ids:
+            parsed_id = parse_resource_id(rid)
+            if len(parsed_id) == 1:
+                logger.error('az resource show: error: argument --ids: invalid ResourceId value: \'%s\'' % rid)
+                return
+            results.append(show_resource(
+                resource_group_name=parsed_id['resource_group'],
+                resource_provider_namespace=parsed_id['resource_namespace'],
+                parent_resource_path=parsed_id['resource_parent'],
+                resource_type=parsed_id['resource_type'],
+                resource_name=parsed_id['resource_name'], api_version=api_version))
+        return results[0] if len(results) == 1 else results
```

Comment:
I do not like this. This means the format of the output will be decided based on the number of input resource id. Since the resource id is represented by a space separated string, it is unknown to the script author if the input value includes one ID or multiple IDs.